ui-modules/blueprint-composer/app/components/providers/blueprint-service.provider.js

```diff
@@ -72,23 +72,26 @@ function BlueprintService($log, $q, $sce, paletteApi, iconGenerator, dslService,
                         config
                             .filter(item => item instanceof Dsl)
                             .reduce((set, config) => {
-                                config.relationships.forEach((entity) => {
-                                    if (entity !== null) {
-                                        set.add({entity: entity, name: key});
-                                    }
-                                });
+                                    config.relationships.forEach((entity) => {
+                                        if (entity !== null) {
+                                            set.add({entity: entity, name: key});
+                                        }
+                                    });
                                 return set;
                             }, set);
                     }
                     if (config instanceof Object) {
                         Object.keys(config)
                             .filter(objectKey => config[objectKey] instanceof Dsl)
                             .reduce((set, objectKey) => {
-                                config[key].relationships.forEach((entity)=> {
-                                    if (entity !== null) {
-                                        set.add({entity: entity, name: key});
-                                    }
-                                });
+                                if(config[key]) {  // when config[objectKey] value is a DSL, but the config does not have a [key] property
+                                    console.log("testing this: " , config[key], key, config, objectKey);
```

Comment:
to remove

Reply:
Yup, forgot the logging statement in there. good catch. :D